src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -194,13 +194,7 @@ object OrderedRDD {
     import kOk._
 
     import Ordering.Implicits._
-
-    rdd.partitioner match {
```

Comment:
Ah, yes, my `preservesPartitioning = true` is definitely wrong (fixed), however, it seems perfectly possible to have one partitioning (which is valid) and want to impose another.  For example, I did a filter and I'm updating tighter bounds.  The code verifies the imposed partitioning, so it is certainly safe.